### DISCUSSION TITLE
Native build for Apple M1 ARM64 architecture

### DISCUSSION THREAD

**Discussion by ultimate-s** [supermerill/SuperSlicer#1119] [Q&A]
Hi there! I was wondering if it's possible to add a build for Apple M1 based systems? I don't know how to compile it and Rosetta2 crashed when trying to use SuperSlicer. Now I have to use Prusaslicer as that now has a native build, but I much rather take my profiles with me to my new m1 based Mac and use SuperSlicer

Can anybody share how they compiled SuperSlicer? I'd like to try if I knew how

**Comment by supermerill** [comment]
I have a report of slow opengl, but that's my first report of crash on m1.
It's planned for 2.3.57, I want a stable version before messing with dependencies.

**Reply by supermerill** [reply]
> Can anybody share how they compiled SuperSlicer? I'd like to try if I knew how

see doc/how to build on mac
https://github.com/supermerill/SuperSlicer/blob/master/doc/How%20to%20build%20-%20Mac%20OS.md

**Reply by rubyredrick** [reply]
For what it's worth.

I tried to use the pre-built dmg, but it won't work on newer MacOS versions because the developer couldn't be verified.

I then followed the how to build on Mac instructions and it runs, AND it built it for M1.

Two things to mention, I did the build on a Mac mini M1, and I am an Apple developer and have Xcode already set up for that, and I don't know whether or not it used my credentials for the certificate.

**Reply by supermerill** [reply]
It certainly uses it.
I don't know how to create a certified mac build, I made some research on it but without a mac, it's hard to do.
If someone can help me to understand how to create a signed dmg, and if possible how to cross-compile to arm, I may be able to update the mac build script to share better mac support.
As it's done remotely, it's a bit hard to give it a key without sharing it with world...
I failed to create a mac VM. I saw another way to have a mac VM on linux, I'll try that when I can.

https://github.com/supermerill/SuperSlicer/issues/1327
